### PR TITLE
[Aftershock] forgotten experiment

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary

SUMMARY: Mods "Adds unique NPC 'Sadie', a forgotten experiment"

#### Purpose of change

Build on Whately storyline, powerful follower, make world feel more fleshed out

#### Describe the solution

New NPC class "forgotten experiment" for Sadie with medical mutant and cyborg traits and items that make some sense for her. Dialog establishing what she knows of who she is, a fetch quest for a new heart, and an intrusive medical service. Bugfix for Wilhemina and Nicodemus dialogue. Bugfix for Cyrus Alpha mission. Bugfix for Cyrus broken cyborg mission. New large storage unit variant that spawns Sadie's nested mapgen. Player can ask Sadie what she thinks of her location using the background talk system and she has some special dialog for some overmap locations and can tell if she's at a safe place or not.

#### Describe alternatives you've considered

Just use an existing class. New missions and effects specific to building locations kicked off by dialog while following. Connection to existing Whately storyline.

#### Testing

Game loads, no errors

Stumbled into Nicodemus and Wilhemina bugs in testing
Screen Shot 2020-10-12 at 8 23 45 AM

Tested them fixed
Screen Shot 2020-10-12 at 9 09 18 AM

Hacking Sadie into existence works, she offers a mission

Screen Shot 2020-10-26 at 9 51 19 PM

The mission spawns Billy as expected

Screen Shot 2020-10-26 at 9 56 35 PM

Sadie can spawn in large storage units (older image, she doesn't have that gear now)

image

Sadie's injection works

image

#### Additional context

Inspired by #37908 and the album Frankenstein Girls Will Seem Strangely Sexy

### CODE REVIEW

File: data/mods/Aftershock/npcs/sadie.json [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    "responses": [
+      { "text": "I'd like to understand more about you first.  Are you human, more or less?", "topic": "TALK_sadie_what" },
+      { "text": "You are aware of the Cataclysm?", "topic": "TALK_sadie_cataclysm" },
+      { "text": "I am your new friend and I'm here to help you.", "topic": "TALK_sadie_1" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_sadie_what",
+    "dynamic_line": "Yes, much of what you see is human I think.  I don't have the proper equipment to be entirely sure and I have some strange memories.  My beacon has already told you that I am Sadie.  Beyond that I don't know much to tell you.  I know a great deal about medicine.  I have some strange responses to physical stimula that I don't completely understand.  I think I must be a medical experiment of some kind.  I am hungry.",
```
Comment (Zireael07) [inline data/mods/Aftershock/npcs/sadie.json:91]: Plural of stimulus is stimul**i**

File: data/mods/Aftershock/npcs/classes.json [hunk 2]
```diff
@@ -159,5 +159,71 @@
...
+  {
+    "type": "npc_class",
+    "id": "afs_forgotten_experiment",
+    "name": "forgotten experiment",
+    "job_description": "What a magical world to explore…",
+    "common": false,
+    "traits": [
+      [ "PRETTY", 100 ],
+      [ "OPTIMISTIC", 100 ],
+      [ "INT_DOWN_3", 100 ],
```
Comment (Maleclypse) [inline data/mods/Aftershock/npcs/classes.json:185]:
```suggestion
      [ "INT_DOWN_3", 100 ],
      [ "FORGETFUL", 100 ],
```
Won't have any real impact on the NPC but thematically it fits.

**Comment by LyleSY** [comment]
weary_test.cpp failure doesn't appear to be related

**Comment by John-Candlebury** [comment]
I'll ask Maleclypse to review this one becuase the more horror segments of Aftershock are more his part.

File: data/mods/Aftershock/npcs/sadie.json [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    "type": "talk_topic",
+    "id": "TALK_sadie_injection2",
+    "dynamic_line": "I'm so glad you liked it.  It really means a lot to me that we have similar body chemistry.  That's so important for all kinds of biological and emotional compatibility for example organ transplants and the reproductive act.  This shows how I feel about you, I think.",
+    "responses": [
+      {
+        "text": "I'm glad we get along so well Sadie.",
+        "effect": [
+          { "u_add_effect": "give_aid", "duration": 2000 },
+          { "u_add_effect": "morale_chat", "duration": 2000 },
+          { "u_add_effect": "sadie_injection", "duration": 2000 }
```
Comment (Maleclypse) [inline data/mods/Aftershock/npcs/sadie.json:193]: Does the duration here use moves or seconds?   I thought it used moves which would mean this effect only lasts twenty seconds.  But I'd need to test it to be certain.

**Review by Maleclypse — APPROVED** [review]
I really like this.  Thank you so much for the work on this!

File: data/mods/Aftershock/npcs/sadie.json (hunk 3 above)
Comment (LyleSY) [reply data/mods/Aftershock/npcs/sadie.json:193]: The NPC doc says “turns”

Comment (LyleSY) [reply data/mods/Aftershock/npcs/sadie.json:193]: If it helps, I built a similar thing for Alonso in the refugee center and his duration is 300    `   "effect": [ { "u_add_effect": "al_relaxed", "duration": 300 }, { "u_consume_item": "condom" } ]
`

Comment (LyleSY) [reply data/mods/Aftershock/npcs/sadie.json:193]: I wanted to be a little more generous here because this is Aftershock, it should be a little extra

Comment (LyleSY) [reply data/mods/Aftershock/npcs/sadie.json]: You were correct, it was gone instantly. Adding two zeros.

File: data/mods/Aftershock/npcs/Backgrounds/BGBR_Sadie.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
+[
+  {
+    "id": "BGSS_Sadie_STORY1",
+    "type": "talk_topic",
+    "dynamic_line": {
+      "and": [
+        {
+          "at_safe_space": true,
+          "no": "This might not be a very private place.",
+          "yes": "This feels like a good place to relax.  Want to play doctor with real scalpels?"
```
Comment (Zireael07) [inline data/mods/Aftershock/npcs/Backgrounds/BGBR_Sadie.json:10]: Can the player say YES to this?

Comment (LyleSY) [reply data/mods/Aftershock/npcs/Backgrounds/BGBR_Sadie.json:10]: No, it’s just flavor text

**Comment by LyleSY** [comment]
Travis error doesn't look related

**Comment by Mrzjadacz** [comment]
I'm unable to make her give me another injection.
After recruiting the option has never appeared.
I successfully completed her mission and received the first one properly.

**Comment by GyprockGypsy** [comment]
Just encountered this character. and was really interested by her story line, but she sent me to a monster that only produced putrid hearts, and which she wont accept.

**Comment by Mrzjadacz** [comment]
> Just encountered this character. and was really interested by her story line, but she sent me to a monster that only produced putrid hearts, and which she wont accept.

You need to make a full butchery or field dress, not dissection
image

**Comment by GyprockGypsy** [comment]
That's not an issue. I just finished reading Bio of a Mountain Man after a bad lab run, and I scummed this spot to try to figure it out. if this was a bug As I said I got the putrid hearts from "Billy" like she wanted, she just wont accept them. I've tried obtaining the hearts through Field Dressing, Full Butchery, and Dissection, and this even clears that stage of the quest marker, but I do not get quest to return to her, and she wont recognize that they are in my inventory. I've also tried doing this with and without the assistance of an NPC.

Win 10
Build 11352
Mods: Aftershock, CRIT, Dino, Magic, Crazy, Map Graphics, Undead People Compatibility Mods

**Comment by Mrzjadacz** [comment]
> That's not an issue. I just finished reading Bio of a Mountain Man after a bad lab run, and I scummed this spot to try to figure it out., I got the putrid hearts from "Billy" like she wanted, she just wont accept them. I've tried obtaining the hearts through Field Dressing, Full Butchery, and Dissection, and this even clears that stage of the quest marker, but I do not get quest to return to her, and she wont recognize that they are in my inventory. I've also tried doing this with and without the assistance of an NPC.
> 
> Win 10
> Build 11352
> Mods: Aftershock, CRIT, Dino, Magic, Crazy, Map Graphics, Undead People Compatibility Mods

You should obtain this item:
image
Report a bug, probably something changed

**Comment by GyprockGypsy** [comment]
Ya, she sent me after a Jabberwock, not a Flesh Golem. Will make an issue.

**Pull request merged**